cachi2/core/package_managers/gomod.py

```diff
@@ -28,6 +29,7 @@
 import pydantic
 import semver
 from packageurl import PackageURL
+from typing_extensions import Self
```

Comment:
this import should be in an `if TYPE_CHECKING` block because typing-extensions are a dev dependency, not a runtime dependency

(they are a runtime dependency of typer and pydantic, but we shouldn't runtime-depend on indirect dependencies)

Reply:
(and Self will need to be "Self")